core/src/main/java/io/opencensus/stats/Aggregate.java

```diff
@@ -0,0 +1,282 @@
+        Function<? super AggregateCount, T> p1,
+        Function<? super AggregateHistogram, T> p2,
+        Function<? super AggregateRange, T> p3,
+        Function<? super AggregateMean, T> p4,
+        Function<? super AggregateStdDev, T> p5) {
+      return p2.apply(this);
+    }
+  }
+
+  /** The range of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class AggregateRange extends Aggregate {
+
+    AggregateRange() {
+    }
+
+    static AggregateRange create(Range range) {
+      return new AutoValue_Aggregate_AggregateRange(range);
+    }
+
+    /**
+     * Returns the aggregated {@code Range}.
+     *
+     * @return the aggregated {@code Range}.
+     */
+    public abstract Range get();
+
+    @Override
+    public final <T> T match(
+        Function<? super AggregateSum, T> p0,
+        Function<? super AggregateCount, T> p1,
+        Function<? super AggregateHistogram, T> p2,
+        Function<? super AggregateRange, T> p3,
+        Function<? super AggregateMean, T> p4,
+        Function<? super AggregateStdDev, T> p5) {
+      return p3.apply(this);
+    }
+  }
+
+  /** The mean value of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class AggregateMean extends Aggregate {
+
+    AggregateMean() {
+    }
+
+    static AggregateMean create(double mean) {
+      return new AutoValue_Aggregate_AggregateMean(mean);
+    }
+
+    /**
+     * Returns the aggregated mean.
+     *
+     * @return the aggregated mean.
+     */
+    public abstract double get();
+
+    @Override
+    public final <T> T match(
+        Function<? super AggregateSum, T> p0,
+        Function<? super AggregateCount, T> p1,
+        Function<? super AggregateHistogram, T> p2,
+        Function<? super AggregateRange, T> p3,
+        Function<? super AggregateMean, T> p4,
+        Function<? super AggregateStdDev, T> p5) {
+      return p4.apply(this);
+    }
+  }
+
+  /** The standard deviation value of aggregated {@code MeasureValue}s. */
+  @Immutable
+  @AutoValue
+  public abstract static class AggregateStdDev extends Aggregate {
+
+    AggregateStdDev() {
+    }
+
+    static AggregateStdDev create(double stdDev) {
+      return new AutoValue_Aggregate_AggregateStdDev(stdDev);
+    }
+
+    /**
+     * Returns the aggregated standard deviation.
+     *
+     * @return the aggregated standard deviation.
+     */
+    public abstract double get();
+
+    @Override
+    public final <T> T match(
+        Function<? super AggregateSum, T> p0,
+        Function<? super AggregateCount, T> p1,
+        Function<? super AggregateHistogram, T> p2,
+        Function<? super AggregateRange, T> p3,
+        Function<? super AggregateMean, T> p4,
+        Function<? super AggregateStdDev, T> p5) {
+      return p5.apply(this);
+    }
+  }
+
+  /** The range of a population's values. */
+  @Immutable
+  @AutoValue
+  public abstract static class Range {
```

Comment:
If we would allow end users to create their own Range, then it should probably go into common. However from my understanding Range should only be created internally, and users should only be allowed to get its value (max and min), and that's why I made its factory method package private.